src/battle.test.c

```diff
@@ -365,18 +365,21 @@ static void test_calculate_armor(CuTest * tc)
     dt.index = 0;
 
     dt.fighter = setup_fighter(&b, du);
-    CuAssertIntEquals_Msg(tc, "default ac", 0, calculate_armor(dt, 0, 0, &magres));
+    CuAssertIntEquals_Msg(tc, "default ac", 0, calculate_armor(dt, 0, 0, select_armor(dt, false), select_armor(dt, true), false));
```

Comment:
Warum wird hier select_armor aufgerufen? Das Ergebnis des Aufrufs sollten wir kennen, oder? Ich hätte die select_armor und select_weapon Funktion lieber in einem eigenen Test geprüft, und hier eine Variable übergeben (z.B. ashield und achain)? Das wäre viel lesbarer, weil man dann nicht erst verstehen muss, wie select_armor funktioniert, um den Test für calulate_armor zu lesen.